nuitka/build/static_src/MainProgram.c

```diff
@@ -1367,6 +1379,81 @@ void setLANGSystemLocaleMacOS(void) {
 #endif
 
 static int Nuitka_Main(int argc, native_command_line_argument_t **argv) {
+
+#if(defined(_NUITKA_LINUX_CONSOLE_MODE_FORCE) || defined(_NUITKA_LINUX_CONSOLE_MODE_DETECT)) && !defined(_WIN32) && !defined(__APPLE__)
+
+if (!isatty(STDIN_FILENO)) {
+
+    const * already_relaunched = getenv("NUITKA_TERMINAL_RELAUNCH");
+    if (already_relaunched == NULL) {
+        setenv("NUITKA_TERMINAL_RELAUNCH", "1", 1);
+
+
+        char
+        const * binary_path = getBinaryFilenameHostEncoded(true);
+
+        char ** spawn_argv = (char ** ) malloc(sizeof(char * ) * (argc + 2));
+        if (spawn_argv == NULL) {
+            fprintf(stderr, "Error: Memory allocation failed for xdg-terminal-exec argv.\n");
+            exit(1); // Exit indicating a failure
+        }
+
+
+        spawn_argv[0] = "xdg-terminal-exec";
```

Comment:
The xdg-terminal-exec should be preferred if it exists, but if it does not, and it may well not do so, we should fall back to invoking something else that mimics it, like `x-terminal-emulator` and such, trying to mimic what `xdg-terminal-exec` does.

Reply:
I am OK with not doing that right away until we are happy with how it works if xdg-terminal-exec exists.